PowerPointLabs/Test/Util/SlideUtil.cs

```diff
@@ -20,25 +20,88 @@ public static void IsSameShape(Shape expShape, Shape actualShape)
+                "different shape height for {0}. exp:{1}, actual:{2}", expShape.Name, expShape.Height, actualShape.Height);
             Assert.IsTrue(expShape.HorizontalFlip == actualShape.HorizontalFlip,
-                "different shape horizontal flip. exp:{0}, actual:{1}", expShape.HorizontalFlip, actualShape.HorizontalFlip);
+                "different shape horizontal flip for {0}. exp:{1}, actual:{2}", expShape.Name, expShape.HorizontalFlip, actualShape.HorizontalFlip);
             Assert.IsTrue(expShape.VerticalFlip == actualShape.VerticalFlip,
-                "different shape vertical flip. exp:{0}, actual:{1}", expShape.VerticalFlip, actualShape.VerticalFlip);
+                "different shape vertical flip for {0}. exp:{1}, actual:{2}", expShape.Name, expShape.VerticalFlip, actualShape.VerticalFlip);
+        }
+
+        public static void IsSameText(Shape expShape, Shape actualShape)
+        {
+            if (expShape.HasTextFrame == Microsoft.Office.Core.MsoTriState.msoTrue)
+            {
+                var expText = expShape.TextFrame.TextRange;
+                var actualText = actualShape.TextFrame.TextRange;
+
+                Assert.IsTrue(expText.Text == actualText.Text,
+                    "different text for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Text, actualText.Text);
+                Assert.IsTrue(expText.Font.Color.RGB == actualText.Font.Color.RGB,
+                    "different font color for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Color.RGB, actualText.Font.Color.RGB);
+                Assert.IsTrue(expText.Font.Bold == actualText.Font.Bold,
+                    "different bold style for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Bold, expText.Font.Bold);
+                Assert.IsTrue(expText.Font.Underline == actualText.Font.Underline,
+                    "different underline style for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Underline, expText.Font.Underline);
+                Assert.IsTrue(expText.Font.Italic == actualText.Font.Italic,
+                    "different italic style for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Italic, expText.Font.Italic);
+                Assert.IsTrue(expText.Font.Size == actualText.Font.Size,
+                    "different text size for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Size, expText.Font.Size);
+            }
         }
 
         public static void IsSameZOrderPosition(Shape expShape, Shape actualShape)
         {
             Assert.IsTrue(expShape.ZOrderPosition == actualShape.ZOrderPosition,
-                "different shape z order. exp:{0}, actual:{1}", expShape.ZOrderPosition, actualShape.ZOrderPosition);
+                "different shape z order for {0}. exp:{1}, actual:{2}", expShape.Name, expShape.ZOrderPosition, actualShape.ZOrderPosition);
+        }
+
+        public static void IsSameShapes(Slide expSlide, Slide actualSlide)
```

Comment:
does this method work for this case:
slide1 has shape A, B, C;
slide2 has shape A, B, D?